src/Plugins/LevelDBStore/Options.cs

```diff
@@ -0,0 +1,213 @@
+        ///
+        /// Default: 1000
+        /// </summary>
+        public int MaxOpenFiles
+        {
+            set { LevelDBInterop.leveldb_options_set_max_open_files(Handle, value); }
+        }
+
+        /// <summary>
+        /// Control over blocks (user data is stored in a set of blocks, and
+        /// a block is the unit of reading from disk).
+        ///
+        /// If not set, leveldb will automatically create and use an 8MB internal cache.
+        /// </summary>
+        public Cache Cache
+        {
+            set
+            {
+                LevelDBInterop.leveldb_options_set_cache(Handle, value.Handle);
+                _Cache = value;
+            }
+            get { return _Cache; }
+        }
+
+        public Comparator Comparator
+        {
+            set
+            {
+                LevelDBInterop.leveldb_options_set_comparator(Handle, value.Handle);
+                _Comparator = value;
+            }
+            get { return _Comparator; }
+        }
+
+        /// <summary>
+        /// Approximate size of user data packed per block.  Note that the
+        /// block size specified here corresponds to uncompressed data.  The
+        /// actual size of the unit read from disk may be smaller if
+        /// compression is enabled.  This parameter can be changed dynamically.
+        ///
+        /// Default: 4K
+        /// </summary>
+        public long BlockSize
+        {
+            set { LevelDBInterop.leveldb_options_set_block_size(Handle, value); }
+        }
+
+        /// <summary>
+        /// Number of keys between restart points for delta encoding of keys.
+        /// This parameter can be changed dynamically.  
+        /// Most clients should leave this parameter alone.
+        ///
+        /// Default: 16
+        /// </summary>
+        public int RestartInterval
+        {
+            set { LevelDBInterop.leveldb_options_set_block_restart_interval(Handle, value); }
+        }
+
+        /// <summary>
+        /// Compress blocks using the specified compression algorithm.  
+        /// This parameter can be changed dynamically.
+        ///
+        /// Default: kSnappyCompression, which gives lightweight but fast compression.
+        ///
+        /// Typical speeds of kSnappyCompression on an Intel(R) Core(TM)2 2.4GHz:
+        ///    ~200-500MB/s compression
+        ///    ~400-800MB/s decompression
+        /// Note that these speeds are significantly faster than most
+        /// persistent storage speeds, and therefore it is typically never
+        /// worth switching to kNoCompression.  Even if the input data is
+        /// incompressible, the kSnappyCompression implementation will
+        /// efficiently detect that and will switch to uncompressed mode.
+        /// </summary>
+        public CompressionLevel CompressionLevel
+        {
+            set { LevelDBInterop.leveldb_options_set_compression(Handle, (int)value); }
```

Comment:
I didn't like only set properties, and I think that I never used a popular library with these kind of properties

Reply:
You can set update the properties at anytime. Which is why its like this.